### DISCUSSION TITLE
Design - Workspace launch actions

### DISCUSSION THREAD

**Discussion by lagacep-ans** [ansys/pyfluent#58] [Ideas]
For icing workspace datamodel, we'll need to replicate what simapp/icing-client is usually doing:
 - Special command line -license="enterprise"
 - Environment variable "PRJAPP_APP=flicing"
 - A scheme command executed on load "(load-fensapice)"
     ... this could be supported by adding some "startup_command" argument to launch_fluent
     
To implement this in pyfluent, two options:
 - Extend launch_fluent( ... , workspace="icing" )
     ... and add to it these automatic arguments/variables/commands
 - Add a launch_workspace ( workspace="icing" , **kwargs )
     ... which will wrap over launch_fluent and add the required arguments
     
The implementation of either option is easy,  the first is adding more complexity in launch_fluent, the second would have less documentation/argument checking if we go through a **kwargs pattern.  Unless we move to a different structure for the launcher settings.

Is there a pattern you would prefer here?  This choice would have long term and user facing implications.

**Comment by mkundu1** [comment]
I'd prefer launch_workspace, perhaps within an ansys/fluent/workspaces directory. If we have workspace-specific code (i.e. when a particular workspace is connected) later on, we can put them in the respective ansys/fluent/workspaces/\<workspace-name\> directory.

**Reply by dnwillia-work** [reply]
Regarding the launching question I agree with Mainak, we should go for the second option.
